### ISSUE TITLE
Failed to attach iso to VM with ConfigDrive on xen/xcpng

### ISSUE THREAD

**Issue by weizhouapache** [apache/cloudstack#7902]
Steps to reproduce the issue
- create network offering with userdata=ConfigDrive
- create a network
- create a vm on the new network
- attach an ISO to the VM

The issue is because there is a ConfigDrive ISO attached to the VM, I cannot attach new ISO.
It works on kvm

```
com.cloud.utils.exception.CloudRuntimeException: Could not find ISO with URL: NFS://10.0.32.4/acs/secondary/ref-trl-5461-x-Mr8-wei-zhou/ref-trl-5461-x-Mr8-wei-zhou-sec1/temp
late/tmpl/1/203/203-1-7c82b718-0c4f-3a35-a89e-a95d188dc81c.iso
        at com.cloud.hypervisor.xenserver.resource.CitrixResourceBase.getIsoVDIByURL(CitrixResourceBase.java:2705)
        at com.cloud.hypervisor.xenserver.resource.XenServerStorageProcessor.attachIso(XenServerStorageProcessor.java:260)
        at com.cloud.hypervisor.xenserver.resource.XenServerStorageProcessor.attachIso(XenServerStorageProcessor.java:91)
        at com.cloud.storage.resource.StorageSubsystemCommandHandlerBase.execute(StorageSubsystemCommandHandlerBase.java:163)
        at com.cloud.storage.resource.StorageSubsystemCommandHandlerBase.handleStorageCommands(StorageSubsystemCommandHandlerBase.java:69)
        at com.cloud.hypervisor.xenserver.resource.wrapper.xenbase.CitrixStorageSubSystemCommandWrapper.execute(CitrixStorageSubSystemCommandWrapper.java:36)
        at com.cloud.hypervisor.xenserver.resource.wrapper.xenbase.CitrixStorageSubSystemCommandWrapper.execute(CitrixStorageSubSystemCommandWrapper.java:30)
        at com.cloud.hypervisor.xenserver.resource.wrapper.xenbase.CitrixRequestWrapper.execute(CitrixRequestWrapper.java:122)
        at com.cloud.hypervisor.xenserver.resource.CitrixResourceBase.executeRequest(CitrixResourceBase.java:1782)
```

##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~

~~~

##### CLOUDSTACK VERSION

~~~

~~~

##### STEPS TO REPRODUCE

~~~

~~~

##### EXPECTED RESULTS

~~~

~~~

##### ACTUAL RESULTS

~~~

~~~

**Comment by weizhouapache** [comment]
I found this issue when tested #7650 on xcpng82

**Comment by sureshanaparti** [comment]
Fixed in #9216
